### PR TITLE
Add Dataflow UI Plugin Utilities

### PR DESCRIPTION
**Pull request by tgroh**

### CODE REVIEW

File: plugins/com.google.cloud.tools.eclipse.dataflow.ui/src/com/google/cloud/tools/eclipse/dataflow/ui/util/SelectFirstMatchingPrefixListener.java [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+
+  public SelectFirstMatchingPrefixListener(Combo targetCombo) {
+    this.targetCombo = targetCombo;
+    lastString = "";
+    contents = ImmutableSortedSet.of();
+    onCompleteListeners = new LinkedHashSet<>();
+  }
+
+  public void setContents(SortedSet<String> contents) {
+    this.contents = contents;
```
Comment (elharo) [inline plugins/com.google.cloud.tools.eclipse.dataflow.ui/src/com/google/cloud/tools/eclipse/dataflow/ui/util/SelectFirstMatchingPrefixListener.java]: safety suggests making a copy here

File: plugins/com.google.cloud.tools.eclipse.dataflow.ui/src/com/google/cloud/tools/eclipse/dataflow/ui/util/SelectFirstMatchingPrefixListener.java [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+  }
+
+  @Override
+  public void modifyText(ModifyEvent e) {
```
Comment (elharo) [inline plugins/com.google.cloud.tools.eclipse.dataflow.ui/src/com/google/cloud/tools/eclipse/dataflow/ui/util/SelectFirstMatchingPrefixListener.java]: e --> event

File: plugins/com.google.cloud.tools.eclipse.dataflow.ui/src/com/google/cloud/tools/eclipse/dataflow/ui/util/SelectFirstMatchingPrefixListener.java (hunk 1 above)
Comment (tgroh) [inline plugins/com.google.cloud.tools.eclipse.dataflow.ui/src/com/google/cloud/tools/eclipse/dataflow/ui/util/SelectFirstMatchingPrefixListener.java]: Done.

File: plugins/com.google.cloud.tools.eclipse.dataflow.ui/src/com/google/cloud/tools/eclipse/dataflow/ui/util/SelectFirstMatchingPrefixListener.java (hunk 2 above)
Comment (tgroh) [inline plugins/com.google.cloud.tools.eclipse.dataflow.ui/src/com/google/cloud/tools/eclipse/dataflow/ui/util/SelectFirstMatchingPrefixListener.java]: Done.

**Comment by tgroh** [comment]
These ones were dirt-simple to add tests for. PTAL

File: plugins/com.google.cloud.tools.eclipse.dataflow.ui.test/src/com/google/cloud/tools/eclipse/dataflow/ui/util/SelectFirstMatchingPrefixListenerTest.java [hunk 3]
```diff
@@ -0,0 +1,100 @@
+package com.google.cloud.tools.eclipse.dataflow.ui.util;
```
Comment (elharo) [inline plugins/com.google.cloud.tools.eclipse.dataflow.ui.test/src/com/google/cloud/tools/eclipse/dataflow/ui/util/SelectFirstMatchingPrefixListenerTest.java]: copyright

Comment (tgroh) [inline plugins/com.google.cloud.tools.eclipse.dataflow.ui.test/src/com/google/cloud/tools/eclipse/dataflow/ui/util/SelectFirstMatchingPrefixListenerTest.java]: Done.

**Pull request merged**